### ISSUE TITLE
Final pyflakes cleanup

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8122]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#8122|
|Type|enhancement|
|Created|2015-11-22 11:57:45Z|
|Branch|https://github.com/twisted/twisted/tree/pyflakes-cleanup-8122|

If and after [#8107](https://github.com/twisted/twisted/issues/8107) is merge we should continue cleaning the remaining pyflakes errors so that tox pyflakes can be run on local computers and get a green light is all is ok.

This ticket should clean the remaining pyflakes errors... some of them being added after the initial cleanup for each submodule :(

<details><summary>Searchable metadata</summary>

```
trac-id__8122 8122
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__None None
branch__branches_pyflakes_cleanup_8122 branches/pyflakes-cleanup-8122
branch_author__adiroiban adiroiban
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1448193465373048 1448193465373048
changetime__1449034610114127 1449034610114127
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

(In [46319]) Branching to pyflakes-cleanup-8122.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban removed owner|
|-|-|

This tries to clean Twisted code of the remaining pyflakes error.

It updates pyflakes as latest version contains a few fixes which affect the twisted code.

twisted/words/xish/xpathparser.py is auto-generated, so there is not much to do. We should just ignore it.

twisted/scripts/tap2deb.py will be deprecated... so until it is removed, just ignore it.

twisted/runner/inetdtap.py is broken and I hope it will be fixed or deprecated in a separate ticket and until then we can ignore it.

There are the remaining pyflakes reports:

```
twisted/words/xish/xpathparser.py:326: redefinition of unused 'sys' from line 36
twisted/words/xish/xpathparser.py:326: redefinition of unused 're' from line 36
twisted/words/xish/xpathparser.py:366: local variable 'END' is assigned to but never used
twisted/words/xish/xpathparser.py:383: local variable 'WILDCARD' is assigned to but never used
twisted/words/xish/xpathparser.py:417: local variable '_context' is assigned to but never used
twisted/words/xish/xpathparser.py:475: local variable '_context' is assigned to but never used
twisted/words/xish/xpathparser.py:485: local variable '_context' is assigned to but never used
twisted/runner/inetdtap.py:22: 'portmap' imported but unused
twisted/runner/inetdtap.py:55: redefinition of unused 'portmap' from line 22
twisted/runner/inetdtap.py:58: undefined name 'rpcVersions'
twisted/runner/inetdtap.py:59: undefined name 'name'
twisted/runner/inetdtap.py:62: undefined name 'p'
twisted/runner/inetdtap.py:152: undefined name 'proto'
twisted/scripts/tap2deb.py:77: local variable 'maintainer' is assigned to but never used
twisted/scripts/tap2deb.py:78: local variable 'description' is assigned to but never used
twisted/scripts/tap2deb.py:80: local variable 'longDescription' is assigned to but never used
twisted/scripts/tap2deb.py:82: local variable 'twistdOption' is assigned to but never used
twisted/scripts/tap2deb.py:83: local variable 'date' is assigned to but never used
twisted/scripts/tap2deb.py:85: local variable 'pythonVersion' is assigned to but never used
```

this branch + changes from [#8107](https://github.com/twisted/twisted/issues/8107) should provide a clean pyflake run.

Please check that changes make sense.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl set owner to @adiroiban|
|-|-|

Hi Adi, a small test, regarding changing the pyflakes version, is failing. Please fix, then merge.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban commented|
|-|-|

Thanks for the review. I fixed the test.

I am not sure what value that test brings... I feel it is on the religious side of TDD :(

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

(In [46382]) Merge pyflakes-cleanup-8122: Clean up the last non-deprecated, non-autogenerated pyflakes warnings

Author: adiroiban
Reviewer: hawkowl
Fixes: [#8122](https://github.com/twisted/twisted/issues/8122)
